types/express-serve-static-core/express-serve-static-core-tests.ts

```diff
@@ -89,14 +91,30 @@ app.get('/nextrouter', (req, res, next) => {
 
 // Default types
 app.post('/', (req, res) => {
-    req.params[0]; // $ExpectType string
+    req.params[0]; // $ExpectError
 
     req.body; // $ExpectType any
     res.send('ok'); // $ExpectType Response<any, Record<string, any>, number>
 });
 
 // Default types - under route
 app.route('/').post((req, res) => {
+    req.params[0]; // $ExpectError
+
+    req.body; // $ExpectType any
+    res.send('ok'); // $ExpectType Response<any, Record<string, any>, number>
+});
+
+// Default types - not using RouteParameters
+app.post('/' as string, (req, res) => {
```

Comment:
This seems like a decent fallback for non standard implementations. Would users still be able to provide custom parameters like in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51262/files#diff-c5d8f61506d81e402ccb7f5f031730ce5ee044b86777af13924d7edfae4243f1R45

Reply:
yeah, this works:
```ts
app.get<{ foo: string; bar: number }>('/' as string, req => {
    req.params.foo; // $ExpectType string
    req.params.bar; // $ExpectType number
    req.params.baz; // $ExpectError
});
```
i can add a test for this if you'd like

Reply:
This is great, good to have this workaround on hand for when it might break someones implementation (although it is hard to imagine how it would). This is a great PR!